pkg/didcomm/dispatcher/outbound/outbound.go

```diff
@@ -382,26 +381,61 @@ func (o *Dispatcher) createForwardMessage(msg []byte, des *service.Destination)
 		routingKeys = des.RoutingKeys
 	}
 
-	// create forward message
-	forward := &model.Forward{
-		Type: forwardMsgType,
-		ID:   uuid.New().String(),
-		To:   des.RecipientKeys[0],
-		Msg:  msg,
+	fwdKeys := append([]string{des.RecipientKeys[0]}, routingKeys...)
+
+	packedMsg, err := o.createPackedNestedForwards(msg, senderKey, fwdKeys, forwardMsgType, mtProfile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create packed nested forwards: %w", err)
+	}
+
+	return packedMsg, nil
+}
+
+func (o *Dispatcher) createPackedNestedForwards(msg, senderKey []byte, routingKeys []string, fwdMsgType, mtProfile string) ([]byte, error) { //nolint: lll
+	for i, key := range routingKeys {
```

Comment:
It turned out that DIDComm v1 and v2 use different order of routingKeys when preparing Forward messages:
- DIDComm v1: direct order (as currently implemented and expected by Verity): https://github.com/hyperledger/aries-rfcs/tree/main/features/0160-connection-protocol#request-transmission
- DIDComm v2: reverse order https://identity.foundation/didcomm-messaging/spec/#sender-process-to-enable-forwarding

It seems we need to support both versions (use the switch above?)